### ISSUE TITLE
Feature request: rebuild bot implementation

### ISSUE THREAD

**Issue by appieschot** [pnp/generator-teams#237]
### 💡 Idea

Rebuild the bot implementation and do something that looks more similar to the BotFx default samples, so it's easier for a dev to copy-paste code from existing samples. 

### Is your feature related to a bug

No

### Additional Info

Once implemented we could do other things like SSO as well

**Comment by stephanbisser** [comment]
I now have a v1 for rebuilding the bot template which is more or less aligned to the official BF sample for a Teams conversation bot [here](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/typescript_nodejs/57.teams-conversation-bot). Any feedback on that?
What I'd like to add in the future here is also the ability to create a bot for Teams based on Adaptive so you can plug it into Composer later on as well but that's maybe a v2 or v3

**Comment by stephanbisser** [comment]
A few things I consider adding to the bot template as well as I think that would make sense:

- Option to enable SSO for the bot (and here I would like to have the generator also to provision the necessary AAD app reg)
- Option to add the Graph client to the bot to access the graph (this would only make sense if SSO is enabled first and then we should decide if we want to add the Graph client by default. So maybe just 1 prompt for SSO & Graph?)

**Comment by RickVanRousselt** [comment]
Indeed a single prompt for both seems normal, why would you else need SSO if not for getting data somewhere. 

For the app reg; Should the generator be provisioning the app reg? On one hand I think for people starting out with Teams dev this is perfect because they can just start and everything is prepped for them, on the other hand if you know what needs to be done, you might already have one available, or have a M365 CLI script that you always reuse. 
So maybe instead of creating a second prompt for the Graph client we need a sub prompt to request if you need the app reg to be provisioned.

**Comment by stephanbisser** [comment]
@RickVanRousselt perfect sounds like a good approach to let people choose if they want to create the app reg on their own or if the generator should do that for you.

@appieschot I guess the CLI could help us creating an app reg and setting all necessary configs to make sure SSO can be established right?

**Comment by wictorwilen** [comment]
Good stuff - for app reg I've been thinking about a couple of things

- generate a file with the required M365 and/or Azure CLI (the former have some of the features available) commands (today it's just outlined in the README file)
- add features to the deploy package to update the bot endpoint using CLI

**Comment by stephanbisser** [comment]
Good points, for the second one we already have an issue in the CLI repo which I started to outline [here](https://github.com/pnp/cli-microsoft365/issues/2529). Now I just need time to add it to the CLI 😂
And for the first point, maybe we can come up with a script as well as a doc in addition to let the generator configure everything on the Azure AD side so people can follow the docs with a script which they can update to fit their needs as well

**Comment by wictorwilen** [comment]
> so people can follow the docs with a script

I always loved the idea of having an option to add Visual Studio Code Tours - that would be really cool if you're running this the first time...

**Comment by stephanbisser** [comment]
I now updated my fork a bit and applied some changes to the folder structure as well as added a couple of files:
![image](https://user-images.githubusercontent.com/29750798/138721453-54365b34-44ae-4921-a75c-8541b21a6379.png)
Now users will be able to add Waterfall or ComponentDialogs quite easily as I already used that for the MainDialog and all the others.

I also added some lines to the server.ts file if the app is going to be a bot to align it what the official botbuilder samples do, like defining the BotFrameworkAdapter as well as the storage in there etc.

I tested that one quite heaviliy but it would be grate if anyone could clone my fork and test if everything works as expected before we merge it into preview?!

Also @wictorwilen would this be then a feature which would result in a new major version (as it implies changes to the scaffolded project)?
